### PR TITLE
fix: fix authorization flow

### LINKED ISSUE #115: Bad request 400 with album
If I want to upload and image with a given url I don't have any problem, but, if I provide an album I get a response 400 (Bad Request) from **node_modules\got\dist\source\as-promise\index.js:117:42**.

First, I thought that it was a credentials issue, instead of using imgur.setClientId(clientId) i used imgur.setCredentials(email, password, clientId) but with this I get RequestError: Missing `url` property from **node_modules\got\dist\source\core\index.js:1372:21**

### PR DESCRIPTION
**Pull request by kaimallea**
The replacment of `request` with `got` required some additional changes. The breakage is what I get for not having proper tests 7 yrs ago. 😅 

fixes #115

### CODE REVIEW

**Comment by kaimallea** [comment]
Adding a test, will merge after I commit it

**Review by KenEucker — APPROVED** [review]
Good fix.

File: lib/mocks/handlers.js [hunk 1]
```diff
@@ -13,6 +13,95 @@ const handlers = [
     };
     return res(ctx.json(response));
   }),
+
```
Comment (kaimallea) [inline lib/mocks/handlers.js:16]: the authorize flow is complex, so the mock for it is too. 😓

**Pull request merged**